### PR TITLE
fix: set nix NIX_KEY_PATH with key data

### PR DESCRIPTION
**Pull request by Pacman99**

### CODE REVIEW

**Comment by nrdxp** [comment]
This very likely broke something, the secrets.sh is not the only pace this value is consumed. Can you give a brief impetus for this so I can understand what you were trying to fix?

File: run/action.yaml [hunk 1]
```diff
@@ -58,9 +58,9 @@ runs:
         DISC_PATH=${{ runner.temp }}/discovery
         DISC_ARC_PATH=${{ runner.temp }}/discovery.tar.zstd
         SSH_AUTH_SOCK=${{ runner.temp }}/ssh_agent.sock
-        NIX_KEY_PATH=${{ inputs.key_path }}
         BUILDER=${{ inputs.builder }}
         CACHE=${{ inputs.cache }}
+        NIX_KEY_PATH=
```
Comment (nrdxp) [inline run/action.yaml:63]: This doesn't actually change anything, if the input is empty this is what would happen anyway

Comment (Pacman99) [reply run/action.yaml:63]: No the input has a default to the runner.temp, specifically a file that doesn't exist unless nix_key is set

**Comment by Pacman99** [comment]
I was trying to fix the error that can occur when nix_key is empty, here is an example https://github.com/input-output-hk/plutus/actions/runs/3710532347/jobs/6290157831

File: run/cache.sh [hunk 2]
```diff
@@ -7,7 +7,7 @@ function upload() {
 
   echo "::debug::uploading$UNCACHED"
 
-  if [[ $CACHE =~ ^s3:// ]]; then
+  if [[ -n "$NIX_KEY_PATH" && $CACHE =~ ^s3:// ]]; then
```
[the remark this answers is not in the document]
Comment (Pacman99) [reply run/cache.sh:10]: Right I realize I did break that, the env from the secrets step doesn't extend to this.

[the remark this answers is not in the document]
Comment (nrdxp) [reply run/cache.sh:10]: Yeah the check is fine, and actually, I'm not sure if the environment is actually extended or not. But it isn't clear in code either way.

File: run/action.yaml (hunk 1 above)
Comment (blaggacao) [reply run/action.yaml:63]: I've to say I also wondered where that default would be coming from, though.

Comment (blaggacao) [reply run/action.yaml:63]: Ah here it is: https://github.com/divnix/std-action/blob/main/run/action.yaml#L8

**Comment by blaggacao** [comment]
I guess even after this change, the code isn't still in its normal form.

**Pull request merged**